File: my_py_pkg/my_py_pkg/my_first_node.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node 

class MyNode(Node):
    def __init__(self):
        super().__init__("py_test")
        self.counter_ = 0 # licznik czasu
        self.get_logger().info("Hello world")
        self.create_timer(1.0, self.timer_callback) #zainicjowanie timera

    def timer_callback(self): #funkcja timera
        self.get_logger().info("Hello aaaa" + str(self.counter_))
        self.counter_ += 1


def main(args=None):
    rclpy.init(args=args)
    # node = Node("py_test")
    # node.get_logger().info("Hello world") #informujaca wiadomosc na poczatku programu
    node = MyNode()
    rclpy.spin(node) #utrzyma "node" aby dzialal do czasu wylaczenia przez ctrl+c
    rclpy.shutdown() #wylacza program


if __name__ == "__main__":
    main()

File: my_cpp_pkg/src/my_first_node.cpp
#include "rclcpp/rclcpp.hpp"

class MyNode : public rclcpp::Node
{
public:
    MyNode() : Node("cpp_test"), counter_(0)
    {
        RCLCPP_INFO(this->get_logger(), "Hello World");
        timer_ = this->create_wall_timer(std::chrono::seconds(1),
                                            std::bind(&MyNode::timerCallback, this));
    }

private:
    void timerCallback()
    {
        RCLCPP_INFO(this->get_logger(), "Hello %d", counter_);
        counter_++;
    }
    rclcpp::TimerBase::SharedPtr timer_;
    int counter_;  
};


int main(int argc, char **argv)
{
    rclcpp::init(argc, argv); // INICJACJA KOMUNIKACJI DLA ROS2
    //auto node = std::make_shared<rclcpp::Node>("cpp_test");
    // RCLCPP_INFO(node->get_logger(), "Hello world"); //wyswietlanie informacji
    
    auto node = std::make_shared<MyNode>(); //projektujemy obiekt
    rclcpp::spin(node);  //#utrzyma "node" aby dzialal do czasu wylaczenia przez ctrl+c
    rclcpp::shutdown();
    return 0;
}